framework/src/main/java/org/apache/drill/test/framework/ConnectionPool.java

```diff
@@ -18,70 +18,57 @@
 package org.apache.drill.test.framework;
 
 import com.google.common.collect.Queues;
-//import org.apache.drill.jdbc.Driver;
-//import com.mapr.drill.jdbc41.Driver;
 import org.apache.log4j.Logger;
 import java.sql.Connection;
 import java.sql.DriverManager;
 import java.sql.SQLException;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Properties;
 import java.util.Queue;
 
-public class ConnectionPool implements DrillDefaults, AutoCloseable {
+public class ConnectionPool implements AutoCloseable {
   private static final Logger LOG = Logger.getLogger(ConnectionPool.class);
-  //private String URL_STRING = "jdbc:drill:drillbit=localhost";
-  //private String JDBC_DRIVER = "org.apache.drill.jdbc.Driver";
-  //private static final String URL_STRING = Utils.getDrillTestProperties().get("CONNECTION_STRING");
-  //private static final String URL_STRING = String.format("jdbc:drill:drillbit=%s",
-  //  Utils.getDrillTestProperties().get("DRILL_STORAGE_PLUGIN_SERVER"));
 
   private final Map<String, Queue<Connection>> connections;
+  private Properties connectionProperties;
 
-  public ConnectionPool(String jdbcDriver) {
+  public ConnectionPool(Properties connectionProperties) {
+    this.connectionProperties = connectionProperties;
     try {
-		Class.forName(jdbcDriver);
-	} catch (ClassNotFoundException e) {
-		e.printStackTrace();
-		System.exit(-1);
-	}
+      Class.forName(DrillTestDefaults.JDBC_DRIVER.getClassName());
+    } catch (ClassNotFoundException e) {
+      e.printStackTrace();
+      System.exit(-1);
+    }
     //Driver.load();
     connections = new HashMap<>();
-    if (jdbcDriver.equals("org.apache.drill.jdbc.Driver")) {
-      TestDriver.driverType = TestDriver.DriverType.APACHE;
-    } else if (jdbcDriver.equals("com.mapr.drill.jdbc41.Driver")) {
-      TestDriver.driverType = TestDriver.DriverType.SIMBA_JDBC;
-    }
   }
 
   public Connection getOrCreateConnection(TestCaseModeler test) throws SQLException {
     final String username = test.matrices.get(0).username;
     final String password = test.matrices.get(0).password;
-    return getOrCreateConnection(username, password);
+    return getOrCreateConnection(username+password);
```

Comment:
You might want to do something more than just concat. I've seen mapr/mapr login and also mapr/maprmapr , so this could be risky. I take it that we usually keep the user and password the same for test environments.. like kuser1/kuser1 , right?

Reply:
Changed key to be username+"|"+password